base/compiler/ssair/ir.jl

```diff
@@ -2,7 +2,7 @@
 
 Core.PhiNode() = Core.PhiNode(Int32[], Any[])
 
-isterminator(@nospecialize(stmt)) = isa(stmt, GotoNode) || isa(stmt, GotoIfNot) || isa(stmt, ReturnNode)
+isterminator(@nospecialize(stmt)) = isa(stmt, GotoNode) || isa(stmt, GotoIfNot) || isa(stmt, ReturnNode) || isa(stmt, EnterNode)
```

Comment:
Should we also add `isa(stmt, Expr) && stmt.head === :leave`?

Leave seems to also be a terminator, but is inconsistently handled.
An an example below we have `is_terminator_or_phi = (isa(stmt, PhiNode) || isterminator(stmt))`.

Reply:
:leave is not a terminator. You can have other things in the same BB

Reply:
Then the comment in L71 is wrong.

Reply:
It is indeed inconsistent - I'm not sure why we did that, but I suspect we can probably remove that code.